feat(configurator): add ButtonConfigurator

leflon opened the pull request:
**Changes of this PR**
This PR adds the `ButtonConfigurator` component, which allows the user to personalize the Pad buttons' properties.

leflon committed:
refactor(pad): add ButtonProperties

Every single button's properties must be stored in the Pad's state. Therefore, we separate the properties that do need to be stored in the state and the props that are only relative to the React Component.

leflon committed:
feat(pad): add `buttonProperties` to PadState

leflon committed:
fix(pad): move `code` to ButtonProperties

leflon committed:
feat(pad): add `selectedButton` to PadState

leflon committed:
feat: add some stuff for ButtonConfigurator

I lost myself in this mess, can't make clean commit..

leflon committed:
feat(pad): add `position` button property

This will allow to locate easily which button was pressed/clicked from the Pad component.

hickatheworld commented:
Note to self:
 - Fix eslint warns
 - Clean the fuck up
 - Try to fix state position
 - Figure out selection and defaults
cya tomorrow

leflon committed:
feat(pad): make buttons selectable with shift

leflon committed:
refactor(pad): always format position to y,x

This avoids confusions.

leflon committed:
fix(pad): always use an array for button positions

better like this. Still formatted as [y, x]

leflon committed:
feat(design): add base ButtonConfigurator design

leflon committed:
fix(configurator): fix change events not firing

leflon committed:
fix(configurator): fix wrong ref name

'resting' color should be renamed to 'idle'

leflon committed:
style(*): remove testing console.logs

leflon committed:
feat(configurator): make input match selection

When the user selected a button to configurate, the configurator's inputs values will update to match the selected button's props.

leflon committed:
feat(design): improve configurator layout

leflon committed:
fix(*): fix eslint warns
